Joedmin-hw3/Joedmin-hw3/Program.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Joedmin_hw3
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                Console.Write("What do you want to do? login/register/exit: ");
+                string answer = Console.ReadLine();
+
+                // Exit
+                if (String.Equals(answer, "exit", StringComparison.OrdinalIgnoreCase))
+                {
+                    Process.GetCurrentProcess().Kill();
+                }
+
+                // Login
+                else if (String.Equals(answer, "login", StringComparison.OrdinalIgnoreCase))
+                {
+                    using (var stream = new StreamReader("Users.txt"))
+                    {
+                        Dictionary<string, string> usersAndPasswords = GetCredentials("Users.txt");
+
+                        string[] credentials = GetCredencials();
+
+                        if (usersAndPasswords.ContainsKey(credentials[0]))
+                        {
+                            if (!(string.Equals(credentials[1], usersAndPasswords[credentials[0]])))
+                                Console.WriteLine("Invalid password.");
+                            else
+                            {
+                                Console.WriteLine("Hello!");
+                            }
+                        }
+                        else
+                            Console.WriteLine("Username is not registered.");
+                    }
+                }
+
+                // Register
+                else if (String.Equals(answer, "register", StringComparison.OrdinalIgnoreCase))
+                {
+                    Dictionary<string, string> usersAndPasswords = GetCredentials("Users.txt");
+                    using (var stream = new StreamWriter("Users.txt", true))
+                    {
+                        string[] credentials = GetCredencials();
+                        SaveToDictionary(usersAndPasswords, credentials);
+                        stream.WriteLine($"{credentials[0]} {credentials[1]}");
+                        Console.WriteLine("Hello!");
+                    }
+                }
+            }
+            catch (UserNotFoundException ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            catch (DuplicateUserCredentialsException ex)
```

Comment:
Hmm...so if the user goes to register.  They are asked to type in a name.  If that name is already in the file then instead of seeing "That name already exists" they will see "The Users.txt contains duplicated users".  It's not bad but it isn't the most obvious / appropriate message.